### DISCUSSION TITLE
Global onError or onSuccess handlers not getting called

### DISCUSSION THREAD

**Discussion by simonknittel** [trpc/trpc#2997] [Q&A]
Hey!

I'm trying to implement global error handling.

I've found posts suggesting that you can do that either via `queryClientConfig.defaultOptions` or via `queryCache`/`mutationCache`.

However, whichever onSuccess, onError, onSettled, onMutate handler I configure globally, none is getting called.

I'm using tRPC v10 with Next.js and implemented it according to https://trpc.io/docs/v10/nextjs

```ts
export const trpc = createTRPCNext<AppRouter>({
  config() {
    return {
      transformer: superjson,

      links: [
        httpBatchLink({
          url: `${getBaseUrl()}/api/trpc`,
        }),
      ],

      queryClientConfig: {
        defaultOptions: {
          queries: {
            onError: (err) => {
              console.log(3, err);
            },
            onSuccess: () => {
              console.log(5);
            },
            onSettled(data, error) {
              console.log(8, data, error);
            },
          },
          mutations: {
            onError: (err) => {
              console.log(2, err);
            },
            onSuccess: () => {
              console.log(6);
            },
            onSettled(data, error) {
              console.log(9, data, error);
            },
          },
        },
        queryCache: new QueryCache({
          onError: (err) => {
            console.log(1, err);
          },
          onSuccess: () => {
            console.log(7);
          },
        }),
        mutationCache: new MutationCache({
          onError: (err) => {
            console.log(256234, err);
          },
          onSuccess: () => {
            console.log(252345);
          },
          onMutate: () => {
            console.log(324634634);
          },
        }),
      },
    };
  },
});
```

Then, I run the query/mutation via `const mutation = trpc.person.add.useMutation();`

The mutations run fine. Just the handlers won't fire. None of the `console.log` shows up in the Developer Console.

Passing the handlers directly to `useMutation({ onError: () => ... })` works fine.

Anything that I'm missing?

**Comment by cnava9389** [comment]
Try looking at https://tanstack.com/query/v4/docs/reference/QueryCache#global-callbacks and then add the queryCache to the queryClient

**Reply by svengau** [reply]
same for me. @simonknittel did you find a solution ? thx
